tests/integration/fix_non_normalized_codes.t

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/perl -w
+
+use ProductOpener::PerlStandards;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Test qw/:all/;
+use ProductOpener::TestDefaults qw/:all/;
+
+use ProductOpener::Config qw/:all/;
+use ProductOpener::Data qw/:all/;
+use ProductOpener::Products qw/:all/;
+use ProductOpener::Store qw/:all/;
+
+remove_all_products();
+wait_application_ready();
+
+sub test_product_path ($code) {
+	my $path = product_path_from_id("$code");
+	$path = $code if $path eq "invalid";
+	return "$data_root/products/$path";
+}
+
+sub retrieve_test_product ($code) {
+	# bare retrieve for test product with strange data
+	my $path = test_product_path($code);
+	return retrieve("$path/product.sto");
+}
+
+# create product in a very minimal way, should be enough for our tests
+sub make_product ($product_ref, $products_collection) {
+	$product_ref->{_id} = $product_ref->{code};
+	my $code = $product_ref->{code};    # copy code to avoid perl to stringify code
+	my $product_path = test_product_path($code);
+	my $rev = $product_ref->{rev};
+	my $only_mongo = delete $product_ref->{only_mongo};
+	if (!$only_mongo) {
+		# use store instead of store_product to avoid normalizations
+		`mkdir -p $product_path`;
+		store("$product_path/$rev.sto", $product_ref);
+		symlink("$rev.sto", "$product_path/product.sto");
+	}
+	# and index in mongo
+	$products_collection->insert_one($product_ref);
+    return;
+}
+
+sub pop_fields ($product_ref, $original_ref) {
```

Comment:
Why is it named pop_fields?

Reply:
Yep, not a really good name, I'll change it.